Sources/Basic/OutputByteStream.swift

```diff
@@ -59,13 +59,13 @@ public class OutputByteStream: OutputStream {
     ///
     /// This method implicitly flushes the stream.
     public var bytes: ByteString {
-        flush()
+        try? flush()
         return ByteString(self.buffer)
     }
     
     // MARK: Data Output API
 
-    public func flush() {
+    public func flush() throws {
         // Do nothing.
```

Comment:
I suggest a better model is one like `fclose()`, where if an error occurred on the stream, it gets captured (and possibly cleared if later things become ok) and thrown from an API which is more convenient to handle (e.g., `close()`).